### PR TITLE
Enable local code build.

### PR DESCRIPTION
**Pull request by frankfliu**
1. Move buildspec.yml to ci folder.
2. Add local container build support.
3. Temporarily disable integration test.

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: ci/Dockerfile.python3.6 [hunk 1]
```diff
@@ -0,0 +1,199 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (lupesko) [inline ci/Dockerfile.python3.6]: There's a lot of duplicity with Dockerfile.python2.7 - can't we use multi stage builds to eliminate duplicity by using a common base Dockerfile?

File: ci/buildspec.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
+# Build Spec for AWS CodeBuild CI
```
Comment (lupesko) [inline ci/buildspec.yml]: License header please

File: ci/buildspec.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+phases:
+  install:
+    commands:
+      - apt-get update
+      - apt-get install -y curl
+      - apt-get install -y libgfortran3
+      - pip install pip -U
+      - pip install flask
+      #- pip install -U mxnet
```
Comment (lupesko) [inline ci/buildspec.yml]: Why is this commented out?

File: ci/buildspec.yml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+      - pip install Pillow
+      - pip install requests
+      - pip install flask-cors
+      - pip install fasteners
+      - pip install jsonschema
+      # - pip install onnx -U
```
Comment (lupesko) [inline ci/buildspec.yml]: Why are these lines commented out?

File: ci/buildspec.yml [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+      # - pip install onnx-mxnet -U
+      - pip install -U -e .
+  build:
+    commands:
+      - python -m pytest mms/tests/unit_tests
```
Comment (lupesko) [inline ci/buildspec.yml]: Are the integration tests taken out intentionally?
Can you please add a comment with a link to an open issue so this does not get missed?

File: ci/dockerd-entrypoint.sh [hunk 6]
```diff
@@ -0,0 +1,23 @@
+#!/bin/sh
```
Comment (lupesko) [inline ci/dockerd-entrypoint.sh]: License header please

Comment (lupesko) [inline ci/dockerd-entrypoint.sh]: Please add a comment explaining how this shell script is intended to be used

File: build.sh [hunk 7]
```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env bash
```
[the remark this answers is not in the document]
Comment (frankfliu) [reply build.sh]: Most of shell scripts in Apache open source project doesn't include license headers. Even aws open source project don't add license headers to shell script, see: https://github.com/aws/aws-codebuild-docker-images/blob/master/ubuntu/ubuntu-base/14.04/dockerd-entrypoint.sh

For upstart script, the header contains service startup information. I'm not sure if it's a good idea to add license header to it. It also makes hard for user to location helpful usage comments from header.

File: build.sh [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+if [ $? -ne 0 ]
+then
+    echo "Please install docker."
+    exit 1
+fi
+
+MMS_HOME="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd )"
+BUILDSPEC="ci/buildspec.yml"
+
+docker pull amazon/aws-codebuild-local:latest --disable-content-trust=false
```
[the remark this answers is not in the document]
Comment (frankfliu) [reply build.sh]: This script is used for developer build locally before push a PR. The local build is identical to code build by using aws-codebuild-local image.

File: ci/Dockerfile.python3.6 (hunk 1 above)
Comment (frankfliu) [reply ci/Dockerfile.python3.6]: We don't update these image often. Merge them won't save us time. And make it more effort to follow aws-codebuild-docker-images updates.

File: ci/Dockerfile.python2.7 [hunk 9]
```diff
@@ -0,0 +1,188 @@
...
+#     http://www.apache.org/licenses/LICENSE-2.0
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+#
+
+FROM ubuntu:14.04.5
+
+ENV LANG="C.UTF-8"
```
[the remark this answers is not in the document]
Comment (frankfliu) [reply ci/Dockerfile.python2.7]: These two sections are copied from aws-codebuild-docker-images: python2.7 and java. I keep it as is to make it easy to update in future.

File: ci/buildspec.yml (hunk 3 above)
Comment (frankfliu) [reply ci/buildspec.yml]: We don't use these dependencies during build.

File: ci/dockerd-entrypoint.sh (hunk 6 above)
Comment (frankfliu) [reply ci/dockerd-entrypoint.sh]: This file is copied from: https://github.com/aws/aws-codebuild-docker-images/blob/master/ubuntu/ubuntu-base/14.04/dockerd-entrypoint.sh

Comment (frankfliu) [reply ci/dockerd-entrypoint.sh]: This file is copied from codebuild repo, better keep it as is.

File: ci/buildspec.yml (hunk 5 above)
Comment (frankfliu) [reply ci/buildspec.yml]: done

**Pull request merged**